src/Core/SegmentMap.cs

```diff
@@ -71,6 +71,17 @@ public SegmentMap(Address addrBase, params ImageSegment[] segments)
 
         public Dictionary<ushort, ImageSegment> Selectors { get; }
 
+        public EndianImageReader CreateImageReader(Address address, IProcessorArchitecture arch)
+        {
+            if(!TryFindSegment(address, out var segment))
+            {
+                throw new InvalidOperationException($"Address {address} not found in program segments");
```

Comment:
This is an argument validation error and should be `ArgumentException`. Also, consider putting a period ('.') at the end of the exception message.